### DISCUSSION TITLE
We need to know the next Seed of the Number Sequence related to Sales Transactions which is going to be used for generating ReceiptId previously to be generated by standard POS in a sales transaction

### DISCUSSION THREAD

**Discussion by rociouchapaz** [microsoft/Dynamics365Commerce.Solutions#227] [Q&A]
We need to know the next Seed of the Number Sequence related to Sales Transactions in Commerce which is going to be used for generating the next ReceiptId previously to be generated by standard POS in a sales transaction

We are using this piece of code in POS side:

 **let numberSequence: any = (window).Commerce.NumberSequence;
 await numberSequence.getNextValueAsync(Commerce.Proxy.Entities.NumberSequenceSeedType.ReceiptSale)
                  .then((transId) =&gt; { receiptId = transId });**

This piece of code is working well, but only in one case it is not working well.

When I deactivate Store Commerce APP and then I activate Store Commerce App, then I have this strange behavior:

When our extension code in POS side is executed we have the next Seed of Number Sequence which is going to be used for generating ReceiptId, but then, in this specific scenario "the first time after activate Store Commerce APP", System doesn't generate the same ReceiptId which previously was generated by our code, the System generates previously ReceiptId which was generated by our code + 1.

In other scenarios, the system generates the same ReceiptId which previously was generated by our code,.

How can we manage this strange behavior? (The first time after activating of Store Commerce App)

**Comment by ahgamal-ms** [comment]
we are able to mitigate this by using getNextReceiptIdAsync

**Reply by rociouchapaz** [reply]
Hi

What is it the difference between getNextReceiptIdAsync and getNextValueAsync?
